test/unit/internal/test_trust.py

```diff
@@ -56,12 +77,117 @@ def test_good(self, asset):
+        [
+            (
+                [_service_v1_op1],
+                [1],
+                ServiceConfiguration(ServiceSelector.ALL),
+                [_service_v1_op1],
+            ),
+            (  # multiple services, same operator: expect 1 service in result
+                [_service_v1_op1, _service2_v1_op1],
+                [1],
+                ServiceConfiguration(ServiceSelector.ALL),
+                [_service2_v1_op1],
+            ),
+            (  # 2 services, different operator: expect 2 services in result
+                [_service_v1_op1, _service_v1_op2],
+                [1],
+                ServiceConfiguration(ServiceSelector.ALL),
+                [_service_v1_op1, _service_v1_op2],
+            ),
+            (  # 3 services, one is not yet valid: expect 2 services in result
+                [_service_v1_op1, _service_v1_op2, _service_v1_op4],
+                [1],
+                ServiceConfiguration(ServiceSelector.ALL),
+                [_service_v1_op1, _service_v1_op2],
+            ),
+            (  # ANY selector: expect 1 service only in result
+                [_service_v1_op1, _service_v1_op2],
+                [1],
+                ServiceConfiguration(ServiceSelector.ANY),
+                [_service_v1_op1],
+            ),
+            (  # EXACT selector: expect configured number of services in result
+                [_service_v1_op1, _service_v1_op2, _service_v1_op3],
+                [1],
+                ServiceConfiguration(ServiceSelector.EXACT, 2),
+                [_service_v1_op1, _service_v1_op2],
+            ),
+            (  # services with different version: expect highest version
+                [_service_v1_op1, _service_v2_op1],
+                [1, 2],
+                ServiceConfiguration(ServiceSelector.ALL),
+                [_service_v2_op1],
+            ),
+            (  # services with different version: expect the supported version
+                [_service_v1_op1, _service_v2_op1],
+                [1],
+                ServiceConfiguration(ServiceSelector.ALL),
+                [_service_v1_op1],
+            ),
+            (  # No supported versions: expect no results
+                [_service_v1_op1, _service_v1_op2],
+                [2],
+                ServiceConfiguration(ServiceSelector.ALL),
+                [],
+            ),
+            (  # services without ServiceConfiguration: expect all supported
+                [_service_v1_op1, _service_v2_op1, _service_v1_op2],
+                [1],
+                None,
+                [_service_v1_op1, _service_v1_op2],
+            ),
+        ],
+    )
+    def test_get_valid_services(self, services, versions, config, expected_result):
+        result = SigningConfig._get_valid_services(services, versions, config)
+
+        assert len(result) == len(expected_result)
+        for s1, s2 in zip(result, expected_result):
+            assert s1.url == s2.url
```

Comment:
List equality should work here, or use `set(result) == set(expected_result)`, if the sorting order is unpredictable.

```suggestion
        assert result == expected_result
```

Reply:
hmm, I was sure some of the protobuf types had issues with `__eq__` but you seem to be right